server/src/main/kotlin/org/jetbrains/bsp/bazel/server/sync/ExecuteService.kt

```diff
@@ -77,20 +79,34 @@ class ExecuteService(
 
     fun test(cancelChecker: CancelChecker, params: TestParams): TestResult {
         val targets = selectTargets(cancelChecker, params.targets)
-        var result = build(cancelChecker, targets, params.originId)
-        if (result.isNotSuccess) {
-            return TestResult(result.statusCode)
-        }
         val targetsSpec = TargetsSpec(targets, emptyList())
 
+        var bazelTestParamsData: BazelTestParamsData? = null
+        try {
+            if (params.dataKind == BazelTestParamsData.DATA_KIND) {
+                val rawParams = params.data.toString()
```

Comment:
I don't know if that's going to work, in other places ([for example the client](https://github.com/JetBrains/intellij-bsp/blob/44cd7cdfdd2d9458dcde94ecfa4c375c47abaab6/src/main/kotlin/org/jetbrains/plugins/bsp/server/client/BspClient.kt#L86)) we use 

```
gson.fromJson(params.data as JsonObject, Class::class.java)
```

Reply:
or rather: if it's going to work, it will probably do a to string/from string roundtrip that could otherwise be avoided